### PR TITLE
Slot to Epoch, and date to slot conversion

### PR DESCRIPTION
**Pull request by nemo83**
Implementing slot to epoch conversion

[x] Shelley slot to epoch conversion
[x] Byron slot to epoch conversion

Implemented date to slot conversion
[x] Shelley date to slot conversion
[x] Byron date to slot conversion

### CODE REVIEW

**Review by matiwinnetou — APPROVED** [review]

File: src/main/java/org/cardanofoundation/conversions/converters/SlotConversions.java [hunk 1]
```diff
@@ -26,4 +27,14 @@ public LocalDateTime slotToTime(long absoluteSlot) {
     // for now post byron we have 1 slot = 1 second
     return genesisConfig.blockTime(EraType.Shelley, absoluteSlot);
   }
+
+    public Long slotToEpoch(long absoluteSlot) {
```
Comment (matiwinnetou) [inline src/main/java/org/cardanofoundation/conversions/converters/SlotConversions.java:31]: Can we somehow indicate that this is partial function in Java?

https://www.pgrs.net/2015/04/23/partial-function-application-in-java-8/

like if this is byron then function doesn't work right..

What do you think?

File: src/main/java/org/cardanofoundation/conversions/exceptioni/UnsupportedConversionException.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package org.cardanofoundation.conversions.exceptioni;
```
Comment (matiwinnetou) [inline src/main/java/org/cardanofoundation/conversions/exceptioni/UnsupportedConversionException.java:1]: why is package called in Italian? exceptioni?

File: src/main/java/org/cardanofoundation/conversions/converters/SlotConversions.java [hunk 3]
```diff
@@ -26,4 +26,26 @@ public LocalDateTime slotToTime(long absoluteSlot) {
...
+
+  /**
+   * Computes the epoch a slot falls in.
+   *
+   * @param absoluteSlot the slot number to convert
+   * @return the Era number the slot falls in
+   */
+  public Long slotToEpoch(long absoluteSlot) {
+    if (absoluteSlot < 0L) {
+      throw new IllegalArgumentException("absoluteSlot cannot be negative");
```
Comment (matiwinnetou) [inline src/main/java/org/cardanofoundation/conversions/converters/SlotConversions.java:38]: sometimes we are using IllegalArgumentException and sometimes InvalidConversionException, can we use 1 or another or these are different cases?

Comment (nemo83) [reply src/main/java/org/cardanofoundation/conversions/converters/SlotConversions.java:38]: gonna ditch the exception and use `IllegalArgumentException` everywhere

Comment (nemo83) [reply src/main/java/org/cardanofoundation/conversions/converters/SlotConversions.java:38]: gonna ditch the exception and use `IllegalArgumentException` everywhere

File: src/main/java/org/cardanofoundation/conversions/exceptioni/UnsupportedConversionException.java (hunk 2 above)
Comment (nemo83) [reply src/main/java/org/cardanofoundation/conversions/exceptioni/UnsupportedConversionException.java]: it was a typo, it's `eccezioni` in italian, `exceptioni` it's just a typo

**Pull request merged**